pkg/pipelines/add_application.go

```diff
@@ -0,0 +1,148 @@
+	GitopsRepo          string
+	GitopsWebhookSecret string
+	Output              string
+	Prefix              string
+	AppGitRepo          string
+	AppWebhookSecret    string
+	AppImageRepo        string
+	EnvName             string
+	DockerCfgJson       string
+	ServicesGitRepo     string
+	SkipChecks          bool
+}
+
+const (
+	overlays     = "overlays"
+	appDir       = "apps"
+	configDir    = "config"
+	servicesDir  = "services"
+	secretName   = "quay"
+	configSApath = "base/config/serviceaccount.yaml"
+	secretPath   = "base/config/secret.yaml"
+	PatchPath    = "overlays/eventlistener_patch.yaml"
+	overlaysDir  = "overlays"
+)
+
+// Note: struct fields must be public in order for unmarshal to
+// correctly populate the data.
+
+type patchStringValue struct {
+	Op    string                          `yaml:"op"`
+	Path  string                          `yaml:"path"`
+	Value triggersv1.EventListenerTrigger `yaml:"value"`
+}
+
+// Add_Application function will initialise the gitops directory
+func Add_Application(o *AddParameters) error {
+
+	if !o.SkipChecks {
+		installed, err := checkTektonInstall()
+		if err != nil {
+			return fmt.Errorf("failed to run Tekton Pipelines installation check: %w", err)
+		}
+		if !installed {
+			return errors.New("failed due to Tekton Pipelines or Triggers are not installed")
+		}
+	}
+
+	gitopsName := getGitopsRepoName(o.GitopsRepo)
+
+	AppRepo := getGitopsRepoName(o.AppGitRepo)
+
+	ServiceRepo := getGitopsRepoName(o.ServicesGitRepo)
+
+	gitopsPath := filepath.Join(o.Output, gitopsName)
+
+	outputs := map[string]interface{}{}
+
+	exists, _ := isExisting(gitopsPath)
+
+	createPatchFiles(outputs, o.EnvName, o.AppGitRepo)
+
+	if !exists {
+		return fmt.Errorf("%s does not exist at %s", gitopsName, gitopsPath)
+	}
+	configPath := filepath.Join(gitopsPath, servicesDir, ServiceRepo)
+
+	files := createResourcesConfig(outputs, o.EnvName, o.DockerCfgJson)
+
+	fileNames, err := writeResources(configPath, files)
+
+	fmt.Println("These are the fileNames", fileNames)
+	if err != nil {
+		return err
+	}
+
+	if err := addKustomize("bases", []string{"../base"}, filepath.Join(gitopsPath, appDir, AppRepo, overlays, kustomize)); err != nil {
+		return err
+	}
+	if err := addKustomize("bases", []string{"../../../services/service-1/overlays"}, filepath.Join(gitopsPath, appDir, AppRepo, baseDir, kustomize)); err != nil {
+		return err
+	}
+	if err := addKustomize("bases", []string{"../config"}, filepath.Join(gitopsPath, servicesDir, ServiceRepo, baseDir, kustomize)); err != nil {
+		return err
+	}
+	if err := addKustomize("bases", []string{"../base"}, filepath.Join(gitopsPath, servicesDir, ServiceRepo, baseDir, kustomize)); err != nil {
+		return err
+	}
+	if err := addKustomize("bases", []string{"../base"}, filepath.Join(gitopsPath, servicesDir, ServiceRepo, overlaysDir, kustomize)); err != nil {
+		return err
+	}
+	kustomize1 := map[string][]string{
+		"bases":     []string{"../../../../envs/chetan-cicd-environment/overlays/"},
```

Comment:
Obviously, fix the hardcoded path

Reply:
You'll want to use `filepath.Rel` for this https://golang.org/pkg/path/filepath/#Rel

Reply:
decided not to be incorporated in this PR.